src/grt/src/MakeWireParasitics.cpp

```diff
@@ -307,6 +367,111 @@ void MakeWireParasitics::makeParasiticsToPin(
   }
 }
 
+void MakeWireParasitics::makePartialParasiticsToPins(
+    std::vector<Pin>& pins,
+    NodeRoutePtMap& node_map,
+    sta::Corner* corner,
+    sta::ParasiticAnalysisPt* analysis_point,
+    sta::Parasitic* parasitic,
+    odb::dbNet* net) const
+{
+  for (Pin& pin : pins) {
+    makePartialParasiticsToPin(
+        pin, node_map, corner, analysis_point, parasitic, net);
+  }
+}
+
+// Make parasitics for the wire from the pin to the grid location of the pin.
+void MakeWireParasitics::makePartialParasiticsToPin(
+    Pin& pin,
+    NodeRoutePtMap& node_map,
+    sta::Corner* corner,
+    sta::ParasiticAnalysisPt* analysis_point,
+    sta::Parasitic* parasitic,
+    odb::dbNet* net) const
+{
+  sta::Pin* sta_pin = staPin(pin);
+  sta::ParasiticNode* pin_node
+      = parasitics_->ensureParasiticNode(parasitic, sta_pin);
+
+  odb::Point pt = pin.getPosition();
+  odb::Point grid_pt = pin.getOnGridPosition();
+
+  std::vector<std::pair<odb::Point, odb::Point>> ap_positions;
+  bool has_access_points = grouter_->pinAccessPointPositions(pin, ap_positions);
+  if (has_access_points) {
+    auto ap_position = ap_positions.front();
+    pt = ap_position.first;
+    grid_pt = ap_position.second;
+  }
+  if (pin.isConnectedToPadOrMacro() || pin.isPort()) {
+    grid_pt = grouter_->findFakePinPosition(pin, net);
+  }
+  // Use the route layer above the pin layer if there is a via
+  // to the pin.
+  int layer = 3;
```

Comment:
Why 3?

Reply:
I define the layers for planar routes as m2 or m3.

Reply:
with layer_h and layer_v beeing 1 (m2) or 2 (m3).

Reply:
And it had to be hardcoded because the it wasn't possible to use the pin access point before layer assignment.